### PR TITLE
Adapted component to be able to read binary QR codes

### PR DESCRIPTION
**Pull request by nitsanshai-clever**
These changes allow the component to read binary QR codes without breaking. The changes replace the AVFoundation framework with ZXingObjC, which under the hood uses AVFoundation. The delegate method now returns an object with the raw bytes of the QR code instead of a string.

These changes likely break other functionality of the component (not tested), but for now we only use the QR reading ability of this component.

The module now relies on the containing React Native Project to have Pods installed with ZXingObjC.

### CODE REVIEW

**Comment by potluck** [comment]
Comments where we made significant changes, else LGTM

**Comment by potluck** [comment]
cc @osdiab

**Comment by potluck** [comment]
recommend looking at the files without whitespace-sensitivity: https://github.com/Clever/react-native-camera/pull/1/files?w=1

File: README.md [hunk 1]
```diff
@@ -1,50 +1,15 @@
-# react-native-camera [![npm version](https://badge.fury.io/js/react-native-camera.svg)](http://badge.fury.io/js/react-native-camera) [![Gitter](https://badges.gitter.im/lwansbrough/react-native-camera.svg)](https://gitter.im/lwansbrough/react-native-camera)
+# react-native-camera
```
Comment (osdiab) [inline README.md]: why does this PR mess around with the README and stuff so much? I'd love to see this ultimately get merged into the mainline `react-native-camera` repo, so might as well keep this stuff as it is there.

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,6 @@
 node_modules
 ios/RCTCamera.xcodeproj/xcuserdata
 ios/RCTCamera.xcodeproj/project.xcworkspace
+
+# OS X
+.DS_Store
```
Comment (osdiab) [inline .gitignore]: Same here for the comment in the README—irrelevant change, I'd just keep it to your own local environment.

File: ios/RCTCamera.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -100,7 +100,7 @@
 		410701251ACB719800C6AA39 /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 0620;
+				LastUpgradeCheck = 0730;
```
Comment (osdiab) [inline ios/RCTCamera.xcodeproj/project.pbxproj]: curious what this means

File: ios/RCTCamera.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -143,6 +143,8 @@
 		410701291ACB719800C6AA39 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				ENABLE_TESTABILITY = YES;
+				ONLY_ACTIVE_ARCH = YES;
```
Comment (osdiab) [inline ios/RCTCamera.xcodeproj/project.pbxproj]: and these

File: README.md (hunk 1 above)
Comment (potluck) [inline README.md]: As in, give `react-native-camera` functionality to scan binary QR codes?

File: package.json [hunk 5]
```diff
@@ -1,12 +1,14 @@
 {
   "name": "react-native-camera",
   "repository": {
-    "type" : "git",
-    "url" : "https://github.com/lwansbrough/react-native-camera.git"
+    "type": "git",
+    "url": "https://github.com/Clever/react-native-camera.git"
+  },
+  "version": "1.0.0",
+  "author": {
```
Comment (osdiab) [inline package.json]: why not just add yourself as a contributor ;)

File: ios/RCTCameraManager.m [hunk 6]
```diff
@@ -760,11 +820,11 @@ - (void)captureOutput:(AVCaptureFileOutput *)captureOutput
   }
 
   NSMutableDictionary *videoInfo = [NSMutableDictionary dictionaryWithDictionary:@{
-     @"duration":[NSNumber numberWithFloat:CMTimeGetSeconds(videoAsAsset.duration)],
-     @"width":[NSNumber numberWithFloat:videoWidth],
-     @"height":[NSNumber numberWithFloat:videoHeight],
-     @"size":[NSNumber numberWithLongLong:captureOutput.recordedFileSize],
-  }];
+                                                                                   @"duration":[NSNumber numberWithFloat:CMTimeGetSeconds(videoAsAsset.duration)],
```
Comment (osdiab) [inline ios/RCTCameraManager.m]: woah, maybe put a newline somewhere to not make this indentation so huge?

File: README.md (hunk 1 above)
Comment (osdiab) [inline README.md]: yeah, unless it's for some reason prohibitively difficult to do so—I guess the fact this uses ZXing to do it would make it difficult to combine the AVFoundation functionality, but it seems strange to wipe mention of the other functionality this library already does.

File: ios/RCTCameraManager.m [hunk 7]
```diff
@@ -14,27 +14,106 @@
...
-  self.session = [AVCaptureSession new];
+  self.capture = [[ZXCapture alloc] init];
+  self.capture.focusMode = AVCaptureFocusModeContinuousAutoFocus;
+
+  self.rctCameraView = [[RCTCamera alloc] initWithManager:self bridge:self.bridge];
+  return self.rctCameraView;
+}
 
-  self.previewLayer = [AVCaptureVideoPreviewLayer layerWithSession:self.session];
+RCT_EXPORT_METHOD(initializeCapture) {
```
Comment (osdiab) [inline ios/RCTCameraManager.m]: I'm a bit out of my element here—I'm finding it hard to parse where along the line this is called and what responsibilities this function has, as it's lacking in comments. Across the board, some documentation would be helpful!

File: ios/RCTCameraManager.m [hunk 8]
```diff
@@ -816,35 +876,34 @@ - (void)captureOutput:(AVCaptureFileOutput *)captureOutput
...
-          }
-        };
-
-        [self.bridge.eventDispatcher sendAppEventWithName:@"CameraBarCodeRead" body:event];
-      }
+- (void)captureResult:(ZXCapture *)capture result:(ZXResult *)result {
+  NSString *type = [self barcodeFormatToString:result.barcodeFormat];
+  NSData *data = [NSData dataWithBytes:result.rawBytes.array length:result.rawBytes.length];
+  NSUInteger bytesCount = data.length;
+  NSString *hexString = @"";
```
Comment (osdiab) [inline ios/RCTCameraManager.m]: I wonder if it makes more sense to return the content as a byte array instead of converting it to a hex string in the library. Byte array is a more accurate type for the data than string, and the client may not care about serializing the data anyway.

File: ios/RCTCameraManager.m [hunk 9]
```diff
@@ -44,7 +44,19 @@ - (UIView *)view
   self.capture.delegate = self;
   self.previewLayer = (AVCaptureVideoPreviewLayer *)self.capture.layer;
   self.previewLayer.needsDisplayOnBoundsChange = YES;
-  [self.rctCameraView.layer addSublayer:self.capture.layer];
+  self.session = self.capture.session;
+  int retries = 0;
+  while (!self.session && retries < 5) {
+    NSLog(@"session was null, retrying");
+    self.capture = [[ZXCapture alloc] init];
```
Comment (potluck) [inline ios/RCTCameraManager.m]: i would pull this out to make the code cleaner

**Comment by alexandersoto** [comment]
@potluck is this PR still valid? Or should we close it since Nitsan left? If not, are you owning?

**Comment by potluck** [comment]
yeah I got it @alexandersoto, will merge it in after syncing the repo with upstream this week
